### PR TITLE
#1403 show all child when no parent selected on visual linked tables

### PR DESCRIPTION
**Pull request by keikeicheung**

### CODE REVIEW

File: vuu/src/main/scala/org/finos/vuu/core/sort/FilterAndSort.scala [hunk 1]
```diff
@@ -16,7 +16,8 @@ case class VisualLinkedFilter(viewPortVisualLink: ViewPortVisualLink) extends Fi
                                         childColumn: Column, source: RowSource, primaryKeys: TablePrimaryKeys): TablePrimaryKeys = {
 
     if (parentSelectionKeys.isEmpty) {
-      InMemTablePrimaryKeys(ImmutableArray.empty[String])
+//      InMemTablePrimaryKeys(ImmutableArray.empty[String])
```
Comment (naleeha) [inline vuu/src/main/scala/org/finos/vuu/core/sort/FilterAndSort.scala:19]: could we remove commented code?

File: vuu/src/test/scala/org/finos/vuu/viewport/VisualLinkedViewPortTest.scala [hunk 2]
```diff
@@ -130,32 +134,48 @@ class VisualLinkedViewPortTest extends AbstractViewPortTestCase with Matchers wi
...
+//      Then("Check we still maintain the selection even with a new sort or filter")
+//      assertVpEqWithMeta(combineQs(viewPortOrders).filter(vpu => vpu.vp.id == viewPortOrders.id)) {
+//        Table(
+//          ("sel"     ,"orderId" ,"trader"  ,"ric"     ,"tradeTime","quantity"),
+//          (0         ,"NYC-0003","chris"   ,"BT.L"    ,1311544800030L,100       ),
+//          (0         ,"NYC-0004","chris"   ,"BT.L"    ,1311544800040L,101       ),
+//          (0         ,"NYC-0005","chris"   ,"BT.L"    ,1311544800050L,102       ),
+//          (0         ,"NYC-0006","chris"   ,"BT.L"    ,1311544800060L,103       )
+//        )
+//      }
```
Comment (naleeha) [inline vuu/src/test/scala/org/finos/vuu/viewport/VisualLinkedViewPortTest.scala:178]: could we clean up commented code if we dont want it or uncomment and keep it? I think its trying to assert that changing the price table selection back to BT.L after sort, still apply the visual link filter on the orders table. Probably could be improved to make it clearer?

File: vuu/src/test/scala/org/finos/vuu/viewport/VisualLinkedViewPortTest.scala [hunk 3]
```diff
@@ -316,5 +336,74 @@ class VisualLinkedViewPortTest extends AbstractViewPortTestCase with Matchers wi
         )
       }
     }
+
+    Scenario("should should all when no parent is selected") {
```
Comment (naleeha) [inline vuu/src/test/scala/org/finos/vuu/viewport/VisualLinkedViewPortTest.scala:340]: think it meant to say "should **show**"

**Comment by heswell** [comment]
I can confirm the problem we were seeing previously noo longer occurs.

**Comment by keikeicheung** [comment]
@naleeha i am going to submit another PR for clearer tests (because i would prefer to split each test only assert one thing to make the test clear)

- could we clean up commented code if we dont want it or uncomment and keep it? I think its trying to assert that changing the price table selection back to BT.L after sort, still apply the visual link filter on the orders table. Probably could be improved to make it clearer?

**Review by naleeha — APPROVED** [review]

**Pull request merged**